### ISSUE TITLE
[Proposal] Model::when() method

### ISSUE THREAD

**Issue by jpkleemans** [laravel/ideas#1975]
## Proposal
Add a static method `when` to `Illuminate\Database\Eloquent\Concerns\HasEvents` to make it easier to hook into (custom) model events: `Model::when(string $event, callable $callback)`.

## Usage example
Let's say we fire a custom event `$this->fireModelEvent('shipped')` in our `Order` model.
Now we can listen to that event and send for example an email notification to the customer:

```php
Order::when('shipped', function ($order) {
    Mail::send(new OrderShippedMail($order));
});
```

**Comment by vinvu-software** [comment]
Good idea, would greatly appreciate this. Been looking for a solution such as this for a while now!
